packages/shared/src/components/squads/SquadPageHeader.tsx

```diff
@@ -122,6 +123,11 @@ export function SquadPageHeader({
               allowedToPost && 'max-w-[30.25rem]',
             )}
             disabled={!allowedToPost}
+            disabledText={
+              isSquadMember
+                ? 'Only admins and moderators can post'
+                : 'Join the squad to create new posts'
```

Comment:
Would still not be valid if the public squad would be set to only admins right?
Just checking for future reference, I know it's an edge-case for now

Reply:
It would not be valid only if user was not a member, since it would always say `Join the squad to create new posts`. But I think that is copy wise because:
1. we might wanna always invite people to join (no matter of squad preferences)
2. we don't expose actual posting permission to anon users (so we could not calculate it)

We would need API adjustments to allow `memberPostingRank` for anon users and check with product about the messaging. I personally think it would just take time but I can pursue if you want.

Reply:
Alright let's leave it as is 👍